src/streaming/HTTPLoader.js

```diff
@@ -0,0 +1,311 @@
+            [HTTPRequest.BITSTREAM_SWITCHING_SEGMENT_TYPE]: ErrorHandler.DOWNLOAD_ERROR_ID_CONTENT,
+            [HTTPRequest.OTHER_TYPE]: ErrorHandler.DOWNLOAD_ERROR_ID_CONTENT
+        };
+    }
+
+    function internalLoad(config, remainingAttempts) {
+
+        let request = config.request;
+        let traces = [];
+        let firstProgress = true;
+        let needFailureReport = true;
+        let requestStartTime = new Date();
+        let lastTraceTime = requestStartTime;
+        let lastTraceReceivedCount = 0;
+        let httpRequest;
+
+        if (!requestModifier || !metricsModel || !errHandler) {
+            throw new Error('config object is not correct or missing');
+        }
+
+        const handleLoaded = function (success) {
+            needFailureReport = false;
+
+            request.requestStartDate = requestStartTime;
+            request.requestEndDate = new Date();
+            request.firstByteDate = request.firstByteDate || requestStartTime;
+
+            if (!request.checkExistenceOnly) {
+                metricsModel.addHttpRequest(
+                    request.mediaType,
+                    null,
+                    request.type,
+                    request.url,
+                    httpRequest.response ? httpRequest.response.responseURL : null,
+                    request.serviceLocation || null,
+                    request.range || null,
+                    request.requestStartDate,
+                    request.firstByteDate,
+                    request.requestEndDate,
+                    httpRequest.response ? httpRequest.response.status : null,
+                    request.duration,
+                    httpRequest.response ? httpRequest.response.responseHeaders || httpRequest.response.getAllResponseHeaders() : null,
+                    success ? traces : null
+                );
+            }
+        };
+
+        const onloadend = function () {
+            if (requests.indexOf(httpRequest) === -1) {
+                return;
+            } else {
+                requests.splice(requests.indexOf(httpRequest), 1);
+            }
+
+            if (needFailureReport) {
+                handleLoaded(false);
+
+                if (remainingAttempts > 0) {
+                    remainingAttempts--;
+                    retryTimers.push(
+                        setTimeout(function () {
+                            internalLoad(config, remainingAttempts);
+                        }, mediaPlayerModel.getRetryIntervalForType(request.type))
+                    );
+                } else {
+                    errHandler.downloadError(
+                        downloadErrorToRequestTypeMap[request.type],
+                        request.url,
+                        request
+                    );
+
+                    if (config.error) {
+                        config.error(request, 'error', httpRequest.http.statusText);
```

Comment:
Should this be `httpRequest.response.statusText`? (although that probably won't be set - see above)